yt_dlp/extractor/common.py

```diff
@@ -1680,12 +1680,16 @@ def traverse_json_ld(json_ld, at_top_level=True):
         traverse_json_ld(json_ld)
         return filter_dict(info)
 
-    def _search_nextjs_data(self, webpage, video_id, *, transform_source=None, fatal=True, **kw):
-        return self._parse_json(
-            self._search_regex(
-                r'(?s)<script[^>]+id=[\'"]__NEXT_DATA__[\'"][^>]*>([^<]+)</script>',
-                webpage, 'next.js data', fatal=fatal, **kw),
-            video_id, transform_source=transform_source, fatal=fatal)
+    def _search_nextjs_data(self, webpage, video_id, *, fatal=True, default=NO_DEFAULT, **kw):
+        # Compat with old usage to make this function non fatal
+        if default == '{}':
+            default = {}
```

Comment:
This is a bugfix and we don't really need this if we can fix the IEs imo. But your call.

Reply:
This might break private plugins though. Maybe fix extractors and use deprecation warning? If you say we dont need it then id like to remove it yeah

Reply:
Sure, deprecation warning's better